tokenizers/src/lib.rs

```diff
@@ -43,101 +38,7 @@ pub fn create_tokenizer_manager(search_tokenizers: Vec<&SearchTokenizer>) -> Tok
-                    .filter(RemoveLongFilter::limit(DEFAULT_REMOVE_TOKEN_LENGTH))
-                    .filter(LowerCaser)
-                    .build(),
-            ),
-            SearchTokenizer::SourceCode => Some(
-                TextAnalyzer::builder(CodeTokenizer::default())
-                    .filter(RemoveLongFilter::limit(DEFAULT_REMOVE_TOKEN_LENGTH))
-                    .filter(LowerCaser)
-                    .filter(AsciiFoldingFilter)
-                    .build(),
-            ),
-            SearchTokenizer::Ngram {
-                min_gram,
-                max_gram,
-                prefix_only,
-            } => Some(
-                TextAnalyzer::builder(
-                    NgramTokenizer::new(*min_gram, *max_gram, *prefix_only).unwrap(),
-                )
-                .filter(RemoveLongFilter::limit(DEFAULT_REMOVE_TOKEN_LENGTH))
-                .filter(LowerCaser)
-                .build(),
-            ),
-            SearchTokenizer::ChineseLindera => Some(
-                TextAnalyzer::builder(LinderaChineseTokenizer::default())
-                    .filter(RemoveLongFilter::limit(DEFAULT_REMOVE_TOKEN_LENGTH))
-                    .filter(LowerCaser)
-                    .build(),
-            ),
-            SearchTokenizer::JapaneseLindera => Some(
-                TextAnalyzer::builder(LinderaJapaneseTokenizer::default())
-                    .filter(RemoveLongFilter::limit(DEFAULT_REMOVE_TOKEN_LENGTH))
-                    .filter(LowerCaser)
-                    .build(),
-            ),
-            SearchTokenizer::KoreanLindera => Some(
-                TextAnalyzer::builder(LinderaKoreanTokenizer::default())
-                    .filter(RemoveLongFilter::limit(DEFAULT_REMOVE_TOKEN_LENGTH))
-                    .filter(LowerCaser)
-                    .build(),
-            ),
-            SearchTokenizer::EnStem => Some(
-                TextAnalyzer::builder(SimpleTokenizer::default())
-                    .filter(RemoveLongFilter::limit(DEFAULT_REMOVE_TOKEN_LENGTH))
-                    .filter(LowerCaser)
-                    .filter(Stemmer::new(Language::English))
-                    .build(),
-            ),
-            SearchTokenizer::Stem { language } => Some(
-                TextAnalyzer::builder(SimpleTokenizer::default())
-                    .filter(RemoveLongFilter::limit(DEFAULT_REMOVE_TOKEN_LENGTH))
-                    .filter(LowerCaser)
-                    .filter(Stemmer::new(*language))
-                    .build(),
-            ),
-            #[cfg(feature = "icu")]
-            SearchTokenizer::ICUTokenizer => Some(
-                TextAnalyzer::builder(ICUTokenizer)
-                    .filter(RemoveLongFilter::limit(DEFAULT_REMOVE_TOKEN_LENGTH))
-                    .filter(LowerCaser)
-                    .build(),
-            ),
-        };
+        let tokenizer_option = search_tokenizer.to_tantivy_tokenizer();
```

Comment:
move previous logic to `to_tantivy_tokenizer` in `SearchTokenizer` for convience